### PR TITLE
Add --missing-help to list command

### PR DESCRIPTION
**Pull request by richsalz**
This is only for master, but what happens after people run this command should be patches for master and 1.1.0 :0

### CODE REVIEW

**Comment by richsalz** [comment]
Pinging @FdaSilvaYY and @ajmohan to take a look. :)

File: apps/openssl.c [hunk 1]
```diff
@@ -296,12 +296,28 @@ static void list_md_fn(const EVP_MD *m,
...
+static void list_missing_help(void)
+{
+    const FUNCTION *fp;
+    const OPTIONS *o;
+
+    for (fp = functions; fp->name != NULL; fp++) {
+        if ((o = fp->help) == NULL)
+            continue;
+        for ( ; o->name != NULL; o++) {
+            if (o->helpstr == NULL)
```
Comment (FdaSilvaYY) [inline apps/openssl.c]: Better test :
`            if (o->helpstr == NULL || o->helpstr[0} == 0) `

Comment (richsalz) [inline apps/openssl.c]: I thought about that but decided against it, in case some app wants to have no extra helpstring.

File: apps/openssl.c [hunk 2]
```diff
@@ -296,12 +296,28 @@ static void list_md_fn(const EVP_MD *m,
     }
 }
 
+static void list_missing_help(void)
+{
+    const FUNCTION *fp;
+    const OPTIONS *o;
+
+    for (fp = functions; fp->name != NULL; fp++) {
+        if ((o = fp->help) == NULL)
```
Comment (ajmohan) [inline apps/openssl.c]: As a function without help is not common, should this function name be printed

File: apps/openssl.c (hunk 1 above)
Comment (ajmohan) [inline apps/openssl.c]: is formatting like this better:

for (int pflag = 0 ; o->name != NULL; o++) {
         if (o->helpstr == NULL) {
                 if (!pflag) {
                         BIO_printf(bio_out, "%s:\n", fp->name);
                         pflag = 1;
                 }       
                 BIO_printf(bio_out, "\t %s\n", o->name);
         }       
}

File: apps/openssl.c (hunk 2 above)
Comment (richsalz) [inline apps/openssl.c]: You're right, every command should at least have a -help :)  Updated commit pushed.

File: apps/openssl.c (hunk 1 above)
Comment (richsalz) [inline apps/openssl.c]: I first had that, but I think the “machine readable” format is better.

Comment (ajmohan) [inline apps/openssl.c]: got it. +1

Comment (richsalz) [inline apps/openssl.c]: Thanks.  Since you did so much work improving the help messages in the past ☺

**Comment by dot-asm** [comment]
+1

@richsalz, I suppose reviewed label was mistake? I mean since you've created the request, you can't mark it as reviewed...

**Comment by richsalz** [comment]
Not quite a mistake.  Since most PR's on github are from non-team members, we require two reviews.  By having team-member code marked as "review" then the other team member does the "second review"  It becomes easier to just look for "second review" things, rather than find the first-review things that were written by someone on the team.

**Comment by richsalz** [comment]
commit 77297115cbaf72533cce99e444542192850e06db.

**Comment by dot-asm** [comment]
> Since most PR's on github are from non-team members, we require two reviews.

But it was *your* PR. Or do you mean that you had one hat on when created request, and another when reviewed???

**Comment by richsalz** [comment]
Code from the team only requires one review.  GitHub non-team code requires two reviews.  To make it "easier" to find PR's that can be merged, I generally make sure the second-review tag is present by making me (or levitte or matt or whoever) as the first reviewer, even tho no review was done.

**Comment by richsalz** [comment]
No.  Team code requires one review.  I did not review the code. I want to easily find things that  can be merged, which means look for the second-review tag.

**Pull request closed without merging**